lib/request_overrider.js

```diff
@@ -307,41 +306,29 @@ function RequestOverrider(req, options, interceptors, remove) {
     if (interceptor.replyFunction) {
       const parsedRequestBody = parseJSONRequestBody(req, requestBody)
 
-      if (interceptor.replyFunction.length === 3) {
+      let fn = interceptor.replyFunction
+      if (fn.length === 3) {
         // Handle the case of an async reply function, the third parameter being the callback.
-        interceptor.replyFunction(
-          options.path,
-          parsedRequestBody,
-          continueWithResponseBody
-        )
-        return
+        fn = util.promisify(fn)
       }
 
-      const replyResponseBody = interceptor.replyFunction(
-        options.path,
-        parsedRequestBody
-      )
-      continueWithResponseBody(null, replyResponseBody)
+      Promise.resolve(fn.call(interceptor, options.path, parsedRequestBody))
```

Comment:
At this point `fn` is either a synchronous function or a promise-returning function; wrapping in `Promise.resolve` makes it into a promise either way.

Is that right? Worth adding as a comment?

Reply:
Correct. Will add.